cxx-squid/src/main/java/org/sonar/cxx/CxxVCppBuildLogParser.java

```diff
@@ -324,155 +351,101 @@ private void ParseCommonCompilerOptions(String line, String fileElement) {
-            line.contains("/G6") ||
-            line.contains("/G3") ||
-            line.contains("/G4")) {
-      AddMacro("_M_IX86", fileElement);    
-    }  
+    }
   }
 
-  private void ParseV120CompilerOptions(String line, String fileElement) {
-    // __AVX__ Defined when /arch:AVX or /arch:AVX2 is specified.
-    // __AVX2__ Defined when /arch:AVX2 is specified.
-    if (line.contains("/arch:AVX ")) {
-      AddMacro("__AVX__", fileElement);    
+  private void ParseV100CompilerOptions(String line, String fileElement) {
+    // VC++ V16.0 - VS2010 (V10.0)
+    AddMacro("__cplusplus=199711L", fileElement);
+    // __cplusplus_winrt Defined when you use the /ZW option to compile. The value of __cplusplus_winrt is 201009.
+    if (line.contains("/ZW ")) {
+      AddMacro("__cplusplus_winrt=201009", fileElement);    
     }
+    AddMacro("_MSC_VER=1600", fileElement);
+    // VS2010 SP1
+    AddMacro("_MSC_FULL_VER=16004021901", fileElement);
+    //_MFC_VER Defines the MFC version. For example, in Visual Studio 2010, _MFC_VER is defined as 0x0C00.
+    AddMacro("_MFC_VER=0x0A00", fileElement);   
+  }
 
-    if (line.contains("/arch:AVX2 ")) {
-      AddMacro("__AVX__", fileElement);
-      AddMacro("__AVX2__", fileElement);    
-    }
-    
-    // _M_ARM Defined for compilations that target ARM processors.
+  private void ParseV110CompilerOptions(String line, String fileElement) {
+    // VC++ V17.0 - VS2012 (V11.0)
+    AddMacro("__cplusplus=199711L", fileElement);
     // __cplusplus_winrt Defined when you use the /ZW option to compile. The value of __cplusplus_winrt is 201009.
     if (line.contains("/ZW ")) {
-      AddMacro("__cplusplus_winrt", fileElement);    
-    }  
-
-    // _CPPUNWIND Defined for code compiled by using one of the /EH (Exception Handling Model) flags.
-    if (line.contains("/EHs ") || 
-            line.contains("/EHa ") || 
-            line.contains("/EHsc ") ||
-            line.contains("/EHac ")) {
-      AddMacro("_CPPUNWIND", fileElement);    
-    } 
+      AddMacro("__cplusplus_winrt=201009", fileElement);    
+    }
+    AddMacro("_MSC_VER=1700", fileElement);
+    // VS2012 Update 4
+    AddMacro("_MSC_FULL_VER=1700610301", fileElement);
+    //_MFC_VER Defines the MFC version. For example, in Visual Studio 2013, _MFC_VER is defined as 0x0C00.
+    AddMacro("_MFC_VER=0x0B00", fileElement);   
+  }
 
-    // _M_ARM_FP Expands to a value indicating which /arch compiler option was used:
-    //    In the range 30-39 if no /arch ARM option was specified, indicating the default architecture for ARM was used (VFPv3).
-    //    In the range 40-49 if /arch:VFPv4 was used.
-    //    See /arch (x86) for more information.       
-    if (line.contains("/arch:IA32 ") || 
-            line.contains("/arch:SSE ") || 
-            line.contains("/arch:SSE2 ") ||
-            line.contains("/arch:AVX2 ") ||
-            line.contains("/arch:AVX ")) {
-      AddMacro("_M_ARM_FP", fileElement);    

```

Comment:
Fine with me because it will have no effect on my analysis. 
Should this go again to ParseV110CompilerOptions(), ParseV120CompilerOptions(), ParseV140CompilerOptions() or shall I add it to  ParseCommonCompilerOptions()?